File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap ={}
        for i in range(len(nums)):
            ans = (target - nums[i])     
            print(ans)
            print(hmap)
            if ans in hmap:
                index = hmap[ans]
                return [index,i]
            else:
                hmap[nums[i]] = i
        
        return []


        [3,4,5,6]
        a = 7 - 3
        a = 4
        h[0] = 3 / a[i]

        a = 7 - 4
        h[1] = 3

File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
     int start = 0 ;
     int end = nums.size() ;
     int mid = (start + end)/2;
     while (start <= end){
        if(nums[mid] == target){
            return mid ;
        }
        else if(nums[mid] > target){
            end = mid - 1 ;
        }
        else if(nums[mid] < target){
            start = mid + 1 ;
        }
        mid = (start + end)/ 2 ;
        cout<<mid<<endl;
     }
     return -1;   
    }
};
